### PR TITLE
fix(unlock-app) No more generate lock address

### PR DESCRIPTION
**Pull request by julien51**
# Description

Since v9 we are not able to generate addresses for locks ahead of time as they are deployed as proxies.
This PR removes that logic and fixes handling of "pending" locks.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/lock/Icon.js [hunk 1]
```diff
@@ -172,8 +172,13 @@ IconModal.propTypes = {
...
   current: PropTypes.string.isRequired,
-  lockAddress: PropTypes.string.isRequired,
-  network: PropTypes.number.isRequired,
+  lockAddress: PropTypes.string,
+  network: PropTypes.number,
+}
+
+IconModal.defaultProps = {
+  lockAddress: '',
+  network: 0,
```
Comment (clemsos) [inline unlock-app/src/components/lock/Icon.js:181]: is that related to this PR?

File: packages/unlock-js/src/unlockService.ts [hunk 2]
```diff
@@ -139,7 +139,9 @@ export default class UnlockService {
       const contractVersion = await contract.publicLockVersion()
       version = parseInt(contractVersion, 10) || 0
     } catch (error) {
-      // This is an older version of Unlock which did not support publicLockVersion
+      console.error(
+        `We could not retrieve the version of the Unlock contract ${address} on this network.`
```
Comment (clemsos) [inline packages/unlock-js/src/unlockService.ts:143]: what is the case where we could not retrieve the lock version? All locks have this, no ?

Comment (julien51) [reply packages/unlock-js/src/unlockService.ts:143]: Generally I think it would be a misconfiguration... and I bumped into that yesterday :) Adding the error makes it not silent!

**Review by clemsos — APPROVED** [review]
lgtm
